src/java/org/apache/cassandra/tcm/RemoteProcessor.java

```diff
@@ -148,6 +148,30 @@ public ClusterMetadata fetchLogAndWait(Epoch waitFor, Retry.Deadline retryPolicy
         }
     }
 
+    @Override
+    public LogState reconstruct(Epoch lowEpoch, Epoch highEpoch, Retry.Deadline retryPolicy)
+    {
+        try
+        {
+            Promise<LogState> request = new AsyncPromise<>();
+            List<InetAddressAndPort> candidates = new ArrayList<>(log.metadata().fullCMSMembers());
+            sendWithCallbackAsync(request,
```

Comment:
should we debounce this?

Reply:
Unlike fetching epoch, I actually expect calls to reconstruct to be less frequent, mostly on bootstraps, and mostly coming from a single node, so I have left a debounce out.